CONTRIBUTING.md

```diff
@@ -0,0 +1,205 @@
+
+### Working on an existing feature
+
+If you choose to work on an existing issue, check out the Rasa contributors board to find existing open issues you can work on.
+
+The issues are handpicked by the Rasa team to have labels which correspond to the difficulty/estimated time needed to resolve the issue.
+
+**To work on an existing issue, go to the contributor project board, add a comment stating you would like to work on it and include any solutions you may already have in mind.**
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/exiting-issue-sara.png)
+
+Someone from Rasa will then assign that issue to you and help you along the way.
+
+---
+
+## How to open a GitHub Pull Request
+
+### What is a Pull Request (PR)?
+
+This is how the GitHub team defines a PR:
+
+> “Pull requests let you tell others about changes you’ve pushed to a branch in a repository on GitHub. Once a pull request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits before your changes are merged into the base branch.”
+
+This process is used by both Rasa team members and Rasa contributors to make changes and improvements to Rasa Open Source.
+
+### What to know before opening a PR
+
+#### Opening issues before PRs
+
+We usually recommend opening an issue before a pull request if there isn’t already an issue for the problem you’d like to solve. This helps facilitate a discussion before deciding on an implementation. See How to open a GitHub issue & file a bug report.
+
+#### Draft PRs
+
+To get some quick initial feedback from the Rasa team, you can create a draft pull request.
+
+#### PRs should be a reasonable length
+
+If your PR is greater than 500 lines, please consider splitting it into multiple smaller contributions.
+
+#### Code style
+
+To ensure a standardized code style we recommend using formatter black. To ensure our type annotations are correct we also suggest using the type checker pytype.
+Travis is a hosted continuous integration service used to build and test software projects hosted at GitHub. If your code is not formatted properly or doesn't type check, Travis will fail to build.
+
+#### Formatting
+
+If you want to automatically format your code on every commit, you can use pre-commit. Just install it via `pip install pre-commit` and execute `pre-commit install` in the root folder. This will add a hook to the repository, which reformats files on every commit.
+
+If you want to set it up manually, install black via `pip install -r requirements-dev.txt.` To reformat files execute `make formatter`.
+
+#### Type Checking
+
+If you want to check types on the codebase, install pytype using `pip install -r requirements-dev.txt`. To check the types execute `make types`.
```

Comment:
They should should also add types and docstrings to any new functions they write!

Reply:
@erohmensing  Is there anywhere I can find more information on this and how we use it?

Reply:
See `Type Annotations` and `Docstrings/Comments` here: https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e